spec/system/admin_branding_spec.rb

```diff
@@ -6,206 +6,266 @@
+        end
+
+        social_media_section_logos.each do |image_type|
+          branding_page.logo_form.remove_image(image_type)
+        end
+
+        branding_page.logo_form.submit
+        expect(branding_page.logo_form).to have_saved_successfully
+
+        social_media_section_logos.each do |image_type|
+          expect(SiteSetting.send(image_type)).to eq(nil)
+        end
       end
+    end
+  end
 
+  describe "fonts" do
+    it "allows to change base and heading font" do
       branding_page.visit
-      branding_page.logo_form.expand_social_media_section
+      branding_page.fonts_form.select_font("base", "helvetica")
 
-      social_media_section_logos.each do |image_type|
-        expect(branding_page.logo_form.image_uploader(image_type)).to have_uploaded_image
-      end
+      expect(branding_page.fonts_form).to have_no_font("heading", "Oswald")
+      branding_page.fonts_form.show_more_fonts("heading")
+      branding_page.fonts_form.select_font("heading", "oswald")
 
-      social_media_section_logos.each do |image_type|
-        branding_page.logo_form.remove_image(image_type)
-      end
+      branding_page.fonts_form.submit
+      expect(branding_page.fonts_form).to have_saved_successfully
+
+      visit "/"
+      branding_page.visit
+      expect(branding_page.fonts_form.active_font("base")).to eq("Helvetica")
+      expect(branding_page.fonts_form.active_font("heading")).to eq("Oswald")
+    end
 
-      branding_page.logo_form.submit
-      expect(branding_page.logo_form).to have_saved_successfully
+    it "allows to change default text size and does not update existing users preferences" do
+      Jobs.run_immediately!
+      branding_page.visit
+      expect(page).to have_css("html.text-size-normal")
+      branding_page.fonts_form.select_default_text_size("larger")
+
+      branding_page.fonts_form.submit
+      expect(modal).to be_open
+      expect(modal.header).to have_content(
+        I18n.t("admin_js.admin.config.branding.fonts.backfill_modal.title"),
+      )
+      modal.close
+      expect(modal).to be_closed
+
+      visit "/"
+      expect(page).to have_css("html.text-size-normal")
+    end
 
-      social_media_section_logos.each do |image_type|
-        expect(SiteSetting.send(image_type)).to eq(nil)
-      end
+    it "allows to change default text size and updates existing users preferences" do
+      Jobs.run_immediately!
+      branding_page.visit
+      expect(page).to have_css("html.text-size-normal")
+      branding_page.fonts_form.select_default_text_size("larger")
+
+      branding_page.fonts_form.submit
+      expect(modal).to be_open
+      expect(modal.header).to have_content(
+        I18n.t("admin_js.admin.config.branding.fonts.backfill_modal.title"),
+      )
+      modal.click_primary_button
+      expect(modal).to be_closed
+
+      visit "/"
+      expect(page).to have_css("html.text-size-larger")
```

Comment:
Same as https://github.com/discourse/discourse/pull/32031/files?diff=split&w=0#r2024409653. We should only run the assertion after the async request has been processed.